### PR TITLE
merging ui to main

### PR DESCRIPTION
**Pull request by WesleyRomanTaylor**

### CODE REVIEW

File: ui/src/app/tag-filter/tag-filter.component.html [hunk 1]
```diff
@@ -10,17 +10,28 @@
...
       (matChipInputTokenEnd)="add($event)">
   </mat-chip-list>
   <mat-autocomplete #auto="matAutocomplete" (optionSelected)="selected($event)">
+<<<<<<< HEAD
     <mat-option *ngFor="let fruit of filteredFruits | async" [value]="fruit">
       {{fruit}}
+=======
+    <mat-option *ngFor="let tag of filteredTags | async" [value]="tag.tag">
+      {{tag.tag}}
+>>>>>>> 661679c (this is a completed view comment page minus the styles)
```
Comment (kyleerin) [inline ui/src/app/tag-filter/tag-filter.component.html:34]: fix these merge conflicts

File: ui/src/app/tag-filter/tag-filter.component.ts [hunk 2]
```diff
@@ -16,10 +27,17 @@ export class TagFilterComponent {
...
   fruitCtrl = new FormControl();
   filteredFruits: Observable<string[]>;
   fruits: string[] = ['Lemon'];
   allFruits: string[] = ['Apple', 'Lemon', 'Lime', 'Orange', 'Strawberry'];
+=======
+  tagCtrl = new FormControl();
+  filteredTags: Observable<TagModel[]>;
+  tags: string[] = [];
+  alltags: TagModel[] = TAG_DATA;
+>>>>>>> 661679c (this is a completed view comment page minus the styles)
```
Comment (kyleerin) [inline ui/src/app/tag-filter/tag-filter.component.ts:40]: merge conflicts

File: ui/src/app/tag-filter/tag-filter.component.ts [hunk 3]
```diff
@@ -59,9 +77,13 @@ export class TagFilterComponent {
...
 
-  private _filter(value: string): string[] {
+  private _filter(value: string): TagModel[] {
     const filterValue = value.toLowerCase();
 
+<<<<<<< HEAD
     return this.allFruits.filter(fruit => fruit.toLowerCase().indexOf(filterValue) === 0);
+=======
+    return this.alltags.filter(tag => tag["tag"].toLowerCase().indexOf(filterValue) === 0);
+>>>>>>> 661679c (this is a completed view comment page minus the styles)
```
Comment (kyleerin) [inline ui/src/app/tag-filter/tag-filter.component.ts:87]: conflicts

**Review by kyleerin — CHANGES_REQUESTED** [review]

File: ui/src/app/tag-filter/tag-filter.component.ts (hunk 3 above)
Comment (WesleyRomanTaylor) [reply ui/src/app/tag-filter/tag-filter.component.ts]: Fixed sorry missed those

File: ui/src/app/tag-filter/tag-filter.component.ts (hunk 2 above)
Comment (WesleyRomanTaylor) [reply ui/src/app/tag-filter/tag-filter.component.ts:40]: fixed

File: ui/src/app/tag-filter/tag-filter.component.html (hunk 1 above)
Comment (WesleyRomanTaylor) [reply ui/src/app/tag-filter/tag-filter.component.html]: fixed

**Review by kyleerin — APPROVED** [review]

**Review by brocwoodworthIBLX — APPROVED** [review]
looks like nothing on the API side was affected, so assuming the UI side is good i'd say LGTM

**Pull request merged**